### PR TITLE
Meta: Add test for feature screenshot

### LINKED ISSUE #5361: Ensure every feature has a compatible screenshot in the readme
The first commit in https://github.com/refined-github/refined-github/pull/5360 passes CI even though some links in the readme are not compatible with the image detection function. We should use the same function in the verifier.

Follows:

- https://github.com/refined-github/refined-github/issues/4698
- https://github.com/refined-github/refined-github/pull/5360

cc @gamemaker1  if you're interested

### PR DESCRIPTION
**Pull request by gamemaker1**
## Related Issues

Closes #5361 if merged

## What This PR Does

Adds a check to the `verify-features` script to ensure every feature has a valid screenshot.

### CODE REVIEW

**Comment by gamemaker1** [comment]
I'm not sure if this is what you want, but this change means both:

1. features that do not have a screenshot, and
2. features that have a screenshot but the url does not match the regex given [here](https://github.com/gamemaker1/refined-github/blob/meta/verify-feature-screenshot/build/readme-parser.ts#L39)

will be flagged as errors

**Comment by cheap-glitch** [comment]
> I'm not sure if this is what you want, but this change means both

We should just check the screenshot URL against the regex. Some features don't have an associated screenshot because they're hard to illustrate.

**Comment by fregante** [comment]
Indeed, but how should we go about it? The parser returns undefined whether the URL is missing or it's not detected as an image.

**Comment by gamemaker1** [comment]
I thought we could write another function that returns the raw markdown from the readme, and then check for the screenshot - but I realised we don't have a list of features that must have a screenshot - so we cannot really check if the right features have valid screenshots.

What we could do instead is create a list of not-allowed file extensions to urls and compare the urls against them - so `.mov` and `.mp4` are not allowed, and any urls with those extensions will be flagged as errors.

What do you think?

**Comment by fregante** [comment]
You're right. There's effectively no difference between an unknown link and an invalid media link. In some cases (for example in hotkeys features) we do have a link, but back to the original issue/PR, not to a screenshot.

I think I found a way:

1. the parser returns the "image URL" that looks like a file, like `/[.][^.]+$/`
2. the checker ensures that `[^.]+$` is either `png` or `gif`.

**Comment by gamemaker1** [comment]
@fregante that worked, thanks!

**Comment by gamemaker1** [comment]
CI will fail on two features that have .mp4 files though

File: build/verify-features.ts [hunk 1]
```diff
@@ -60,6 +60,10 @@ function findError(filename: string): string | void {
 		return `ERR: ${featureId} should be described better in the readme (at least 20 characters)`;
 	}
 
+	if (featureMeta.screenshot && !/\.(png|jpg|gif)$/.test(featureMeta.screenshot)) {
```
Comment (fregante) [inline build/verify-features.ts:63]: Screenshots should not be jpegs

**Review by fregante — COMMENTED** [review]
Looks good! Just one change

File: build/verify-features.ts (hunk 1 above)
Comment (gamemaker1) [reply build/verify-features.ts]: Done, but the following three features have a .jpg image:

```
ERR: comment-on-draft-pr-indicator should have a screenshot (png/gif) in the readme
ERR: link-to-github-io should have a screenshot (png/gif) in the readme
ERR: one-click-pr-or-gist should have a screenshot (png/gif) in the readme
```

Comment (yakov116) [reply build/verify-features.ts]: Can you download them and reload them and just change the extension before uploading it.

Comment (gamemaker1) [reply build/verify-features.ts]: Changed the extension, but the files are still jpegs

Comment (gamemaker1) [reply build/verify-features.ts]: Converted the files to pngs 🎉

**Review by fregante — APPROVED** [review]

**Comment by fregante** [comment]
Thanks again @gamemaker1!

**Pull request merged**